rpc/query_server.go

```diff
@@ -599,6 +614,35 @@ func (s *QueryServer) GetContractRecord(contractAddrStr string) (*types.Contract
 	return k, nil
 }
 
+func (s *QueryServer) DposTotalStaked() (*types.DposTotalStakedResponse, error) {
+	snapshot := s.StateProvider.ReadOnlyState()
+	defer snapshot.Release()
+	if s.totalStakedAmount != nil {
+		if time.Since(s.totalStakedAmount.createAt) <= time.Minute {
+			return &types.DposTotalStakedResponse{
+				TotalStaked: &s.totalStakedAmount.amount,
+			}, nil
+		}
+	}
+
+	dposCtx, err := s.createDposV3Ctx(snapshot)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := dposv3.TotalStaked(dposCtx, s.BootstrapNodes)
+	if err != nil {
+		return nil, err
+	}
+	s.totalStakedAmount = &totalStakedAmount{
+		createAt: time.Now(),
+		amount:   *resp,
+	}
+
+	return &types.DposTotalStakedResponse{
```

Comment:
This RPC method should return a simple JSON response, not a protobuf

Reply:
Done